lib/mongo/operation/read_preferrable.rb

```diff
@@ -24,7 +24,8 @@ module ReadPreferrable
 
       def update_selector(context)
         if context.mongos? && read_pref = read.to_mongos
-          selector.merge(:$readPreference => read_pref)
+          { :$query => selector,
```

Comment:
What if the selector already has $query in it because of some other $operator ($orderby, $maxTimeMS, etc.)?